test/rules/prefer_final_locals_test.dart

```diff
@@ -112,74 +107,88 @@ f() {
     await assertDiagnostics(r'''
 f() {
   var (a, b) = ('a', 'b');
-  print('$a$b');
 }
 ''', [
-      lint(21, 10),
+      lint(8, 3),
     ]);
   }
 
-  test_destructured_recordPattern_list() async {
+  test_nonDeclaration_destructured_recordPattern() async {
+    await assertNoDiagnostics(r'''
+f(String a, String b) {
+  [a, b] = ['a', 'b'];
+}
+''');
+  }
+
+  test_destructured_recordPattern_withParenthesizedPattern() async {
+    await assertDiagnostics(r'''
+f() {
+  var ((a, b)) = ('a', 'b');
+}
+''', [
+      lint(8, 3),
+    ]);
+  }
+
+  test_destructured_listPattern() async {
     await assertDiagnostics(r'''
 f() {
   var [a, b] = ['a', 'b'];
-  print('$a$b');
 }
 ''', [
-      lint(12, 6),
+      lint(8, 3),
```

Comment:
I'm having a hard time understanding this diff...  Are the test expectations changing here?

Reply:
`[12, 6]` is the range for `[a, b]`, the ListPattern. `[8, 3]` is the range for `var`. This change keeps `PatternVariableDeclaration`s in sync with `VariableDeclarationList`s. I think this is a positive change, but I haven't thought too much of more complicated examples...

Reply:
I think this is the right direction. I'm doing some spot checking and will confirm. Thanks!